scripts/collisions.js

```diff
@@ -59,26 +66,36 @@ const worker = (methodName, done) => {
   const stats = new Stats()
   let hasCollided = false
   let sum = 0
+  let startTime = Date.now()
+  let numRuns = 0
 
-  const isComplete = () => {
-    const moe = stats.length > 2
+  const computeMoe = () => stats.length > 2 && stats.amean() > 0
       ? stats.moe() / stats.amean()
       : null
 
-    return stats.length >= MIN_RUNS && (moe != null && (moe <= LO_MOE || (moe <= HI_MOE && sum >= MAX_SUM)))
+  const computeDuration = () => Date.now() - startTime
+
+  const isComplete = () => {
+    const duration = computeDuration()
+
+    const moe = computeMoe()
+    return numRuns >= MIN_RUNS &&
+      ((duration >= MAX_DURATION) ||
+       ((moe != null && (moe <= LO_MOE || (moe <= HI_MOE && sum >= MAX_SUM))) || (moe == null && !hasCollided)))
   }
 
   while (!isComplete()) {
     const firstCollisionN = findFirstCollisionN(methodName)
 
-    if (findFirstCollisionN != null) {
+    if (firstCollisionN != null) {
       hasCollided = true
       sum += firstCollisionN
+      stats.push(firstCollisionN)
     } else {
       sum = MAX_N
     }
 
-    stats.push(firstCollisionN)
```

Comment:
We shouldn't be pushing `null` results (which happens if we found no collisions and reached `MAX_N`), since there isn't a datapoint to work with in this case. This also means we now need to maintain our own separate counter (`numRuns`)